### PR TITLE
Update tests and examples

### PR DESCRIPTION
**Pull request by msimberg**
- Make more tests and examples compile with the local runtime
- Update the OpenMP examples and enable them in one CI configuration
- Move some more tests to modules (functional)

### CODE REVIEW

File: examples/1d_stencil/1d_stencil_1.cpp [hunk 1]
```diff
@@ -105,32 +106,32 @@ int hpx_main(hpx::program_options::variables_map& vm)
...
+        "Local x dimension") ("nt",
+        po::value<std::uint64_t>()->default_value(45),
+        "Number of time steps") ("k", po::value<double>(&k)->default_value(0.5),
+        "Heat transfer coefficient (default: 0.5)") ("dt",
+        po::value<double>(&dt)->default_value(1.0),
+        "Timestep unit (default: 1.0[s])") ("dx",
+        po::value<double>(&dx)->default_value(1.0),
+        "Local x dimension") ("no-header",
+        "do not print out the csv header row");
+    // clang-format on
```
Comment (hkaiser) [inline examples/1d_stencil/1d_stencil_1.cpp:130]: Can we avoid reformatting the program options, please?

File: examples/jacobi_smp/jacobi.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include <iostream>
 #include <string>
 
+using hpx::program_options::command_line_parser;
 using hpx::program_options::options_description;
 using hpx::program_options::parse_command_line;
```
Comment (hkaiser) [inline examples/jacobi_smp/jacobi.cpp:20]: This is not needed anymore, I think.

File: examples/quickstart/customize_async.cpp [hunk 3]
```diff
@@ -79,5 +79,10 @@ int main()
         f.wait();
     }
 
-    return 0;
+    return hpx::local::finalize();
+}
+
+int main(int argc, char* argv[])
```
Comment (hkaiser) [inline examples/quickstart/customize_async.cpp:85]: Unrelated to this PR: should we add support for `#include <hpx/local/hpx_main.hpp>`?

File: examples/1d_stencil/1d_stencil_1.cpp (hunk 1 above)
Comment (msimberg) [reply examples/1d_stencil/1d_stencil_1.cpp:130]: Yes. I've tried to them from clang-formatting but missed this one. Thanks!

File: examples/quickstart/customize_async.cpp (hunk 3 above)
Comment (msimberg) [reply examples/quickstart/customize_async.cpp:85]: My preference is not to add it.

Comment (hkaiser) [reply examples/quickstart/customize_async.cpp:85]: What's your rationale for this? Wouldn't it simplify porting codes?

Comment (msimberg) [reply examples/quickstart/customize_async.cpp:85]: Not noticeably. While it's convenient on the surface, the implementation is complex, error prone and not really portable. That's of course only my opinion and I can see how others might think that's a fine tradeoff.

**Review by aurianer — DISMISSED** [review]
LGTM thanks!

**Pull request merged**